File: tests/integration/octave_modular/preprocessor.m
function y = preprocessor(filename)
	init_shogun;
	y=true;

	addpath('util');
	addpath('../data/preprocessor');

	eval('globals'); % ugly hack to have vars from filename as globals
	eval(filename);

	if strcmp(preprocessor_name, 'LogPlusOne')==1
		preproc=LogPlusOne();
	elseif strcmp(preprocessor_name, 'NormOne')==1
		preproc=NormOne();
	elseif strcmp(preprocessor_name, 'PruneVarSubMean')==1
		preproc=PruneVarSubMean(tobool(preprocessor_arg0_divide));
	elseif strcmp(preprocessor_name, 'SortUlongString')==1
		preproc=SortUlongString();
	elseif strcmp(preprocessor_name, 'SortWordString')==1
		preproc=SortWordString();
	else
		error('Unsupported preproc %s', preprocessor_name);
	end

	if ~set_features('kernel_')
		return;
	end

	preproc.init(feats_train);
	feats_train.add_preprocessor(preproc);
	feats_train.apply_preprocessor();
	feats_test.add_preprocessor(preproc);
	feats_test.apply_preprocessor();

	if ~set_kernel()
		return;
	end

	km_train=max(max(abs(kernel_matrix_train-kernel.get_kernel_matrix())));
	kernel.init(feats_train, feats_test);
	km_test=max(max(abs(kernel_matrix_test-kernel.get_kernel_matrix())));

	data={'kernel', km_train, km_test};
	y=check_accuracy(kernel_accuracy, data);

File: tests/integration/python_modular/preprocessor.py
"""
Test Preprocessor
"""

from modshogun import *

import util


########################################################################
# kernel computation
########################################################################

def _evaluate (indata):
	prefix='kernel_'
	feats=util.get_features(indata, prefix)
	kfun=eval(indata[prefix+'name']+'Kernel')
	kargs=util.get_args(indata, prefix)

	prefix='preprocessor_'
	pargs=util.get_args(indata, prefix)
	feats=util.add_preprocessor(indata[prefix+'name'], feats, *pargs)

	prefix='kernel_'
	kernel=kfun(feats['train'], feats['train'], *kargs)
	km_train=max(abs(
		indata[prefix+'matrix_train']-kernel.get_kernel_matrix()).flat)
	kernel.init(feats['train'], feats['test'])
	km_test=max(abs(
		indata[prefix+'matrix_test']-kernel.get_kernel_matrix()).flat)

	return util.check_accuracy(
		indata[prefix+'accuracy'], km_train=km_train, km_test=km_test)


########################################################################
# public
########################################################################

def test (indata):
	return _evaluate(indata)

